biojava-structure/src/main/java/org/biojava/nbio/structure/asa/AsaCalculator.java

```diff
@@ -317,36 +361,118 @@ public AsaCalculator(Point3d[] atomCoords, double probe, int nSpherePoints, int
 	}
 
 	/**
-	 * Returns list of indices of atoms within probe distance to atom k.
-	 * @param k index of atom for which we want neighbor indices
+	 * Returns the 2-dimensional array with neighbor indices for every atom.
+	 * @return 2-dimensional array of size: n_atoms x n_neighbors_per_atom
 	 */
-	private Integer[] findNeighborIndices(int k) {
+	int[][] findNeighborIndices() {
+
 		// looking at a typical protein case, number of neighbours are from ~10 to ~50, with an average of ~30
-		// Thus 40 seems to be a good compromise for the starting capacity
-		ArrayList<Integer> neighbor_indices = new ArrayList<>(40);
+		int initialCapacity = 60;
 
-		double radius = radii[k] + probe + probe;
+		int[][] nbsIndices = new int[atomCoords.length][];
 
-		for (int i=0;i<atomCoords.length;i++) {
-			if (i==k) continue;
+		for (int k=0; k<atomCoords.length; k++) {
+			double radius = radii[k] + probe + probe;
+
+			List<Integer> thisNbIndices = new ArrayList<>(initialCapacity);
+
+			for (int i = 0; i < atomCoords.length; i++) {
+				if (i == k) continue;
+
+				double dist = atomCoords[i].distance(atomCoords[k]);
+
+				if (dist < radius + radii[i]) {
+					thisNbIndices.add(i);
+				}
+			}
+
+			int[] indicesArray = new int[thisNbIndices.size()];
+			for (int i=0;i<thisNbIndices.size();i++) indicesArray[i] = thisNbIndices.get(i);
```

Comment:
Very minor note. Since random access isn't being used in the array list, could use Queue interface instead of List.

Reply:
Some cute shortcuts - https://stackoverflow.com/questions/960431/how-to-convert-listinteger-to-int-in-java . I don't know if biojava has any general preference regarding guava or apache commons.

Reply:
Thanks! I like the answer that uses the stream API: https://stackoverflow.com/a/23945015/3914327